bazel_tools/build_environment.bzl

```diff
@@ -12,11 +21,17 @@ def _impl(ctx):
     # Generates a simple Bazel file that just sets a bunch of Bazel variables,
     # so they can be used in our main Bazel BUILD files.
     semver = ctx.os.environ.get("DAML_SDK_RELEASE_VERSION", default = "0.0.0")
-    if semver.find("-snapshot.") > 0:
-        ghc = semver[:-9].replace("-snapshot.", ".")
+    if semver.find("-") > 0:
+        ghc_list = []
+        for segment in semver.replace("-", ".").split("."):
+            if is_number(segment):
+                ghc_list.append(segment)
+        ghc = ghc_list[0]
+        for elem in ghc_list[1:]:
+            ghc = ghc + "." + elem
```

Comment:
```suggestion
        ghc = ".".join(ghc_list)
```

Reply:
I was looking for a method on list. 🤦

Reply:
IMO Python is the only language that does this "correctly". :stuck_out_tongue: